var/spack/repos/builtin/packages/goshimmer/package.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Goshimmer(Package):
+    """GoShimmer is a prototype node for exploring the implementation of IOTA 2.0"""
+    homepage = 'https://github.com/iotaledger/goshimmer'
+    url      = 'https://github.com/iotaledger/goshimmer/archive/refs/tags/v0.8.0.tar.gz'
+
+    maintainers = ['bernhardkaindl']
+
+    version('0.8.0', 'ec515deebf0dd35ff76cd98addae9cfcbf4758ab9eb72eb0f6ff4654f2658980')
+
+    depends_on('go@1.16:', type='build')
+    depends_on('snappy')
+    depends_on('rocksdb')
+    depends_on('zstd')
+    depends_on('zlib')
+    depends_on('lz4')
+
+    @property
+    def snapbin(self):
+        return join_path(self.prefix.bin, 'snapshot.bin')
+
+    def patch(self):
+        for file in ['tools/genesis-snapshot/main', 'plugins/messagelayer/parameters']:
+            filter_file('./snapshot.bin', self.snapbin, file + '.go')
+
+    def install(self, spec, prefix):
+        which('go')('build', '-modcacherw', '-tags', 'rocksdb,netgo')
+        mkdir(prefix.bin)
+        install('config.default.json', prefix.bin)
+        install('goshimmer',           prefix.bin)
+        which('wget')('-O', self.snapbin,
+                      'https://dbfiles-goshimmer.s3.eu-central-1.amazonaws.com/snapshots/nectar/snapshot-latest.bin')
```

Comment:
Can this change over time? Generally it is ideal for an installation to be reproducible, so if this can be updated that might make things confusing.

Reply:
@bernhardkaindl is there a way to download something fixed rather than `snapshot-latest`?

Reply:
No, as far as I understand, it's a data file for joining a network, holding the network state.

The download could better happen using a wrapper script on first invocation,
but since the project I was building it for is not going to use the goshimmer project for now,
I'm not testing or maintaining this for now.

The software is "only" a research prototype for the next-gen IOTA 2.0 protocol,
it's not for production use:
* We can close this PR, if you'd rather tend to object merging it as is.
* Or you can unassign yourself and I keep can keep it for future work to pick up later.

Reply:
I can merge this as-is: from reading the installation instructions on the home page this is part of the official installation process (it looked odd to me though at first).